app/models/maintenance_tasks/run.rb

```diff
@@ -206,6 +208,24 @@ def csv_attachment_presence
       nil
     end
 
+    # Performs validation on the arguments to use for the Task. If the Task is
+    # invalid, the errors are added to the Run.
+    def valid_task_arguments?
+      if task.invalid?
+        error_messages = []
+        task.errors.messages.each do |attribute, messages|
+          error_messages +=
+            messages.map { |message| "#{attribute.inspect} #{message}" }
+        end
```

Comment:
This is unfortunately the only way to do it that is compatible with both Rails 6.0 and Rails 6.1 (since in Rails 6, error messages are [yielded as hashes](https://api.rubyonrails.org/v6.0/classes/ActiveModel/Errors.html#method-i-each))

Reply:
Good catch.

Reply:
Better to define the methods conditionally based in the Rails version with different implementations so we know we can remove the 6.0 implementation when 6.0 is not supported anymore.